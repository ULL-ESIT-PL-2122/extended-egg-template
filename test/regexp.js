let insp = require("util").inspect;
let ins = (x) => insp(x, {depth:null});
let exec = require('child_process').exec;
var should = require("should");
let e2t = require('@ull-esit-pl/example2test');

describe("Regular Expressions in Egg", function() {
  let runTest = (programName, done) => {
    debugger;
    e2t({
      exampleInput: programName+'.egg', 
      executable: 'bin/egg.js', 
      assertion: (result, expected) => result.replace(/\s+/g,'').
      should.eql(expected.replace(/\s+/g,'')),

      done: done, 
    });
  };

  it("testing regexp.egg", function(done) {
    runTest('regexp', done);
  });


  it("testing regexp-simple.egg", function(done) {
    runTest('regexp-simple', done);
  });

  it("testing regexp-global.egg", function(done) {
    runTest('regexp-global', done);
  });

  it("testing regexp-escape.egg", function(done) {
    runTest('regexp-escape', done);
  });

  it("testing regexp-bracket.egg", function(done) {
    runTest('regexp-bracket', done);
  });
});
