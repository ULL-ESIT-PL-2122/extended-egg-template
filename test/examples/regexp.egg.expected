true
["1987-07-14","1987","07","14"]
0
["2015-02-22","2015","02","22"]
2015
02
